src/Engine/EngineApp/EngineApp.cpp

```diff
@@ -3,34 +3,37 @@
 #include <time.h>
 #include <iostream>
 
-EngineApp::EngineApp() {
-    srand((unsigned int)(time(0)));
-    RESOURCE_DIR = "../resources/";
-    APP_NAME = "";
-    verbose = false;
-    nFrames = fps = 0;
-    timeStep = lastFrameTime = runTime = 0.0;
-}
+
+std::string EngineApp::RESOURCE_DIR = "../resources/";
+std::string EngineApp::APP_NAME = "Battle Royale with Cheese";
+int EngineApp::fps = 0;
+double EngineApp::timeStep = 0.0;
+bool EngineApp::verbose = false;
+double EngineApp::lastFpsTime = 0.0;
+int EngineApp::nFrames = 0;
+double EngineApp::lastFrameTime = 0.0;
+double EngineApp::runTime = 0.0;
 
 int EngineApp::init() {
-    if (windowHandler.init(APP_NAME)) {
+    srand((unsigned int)(time(0)));    
+    if (Window::init(APP_NAME)) {
         return 1;
     }
 
-    m_scene.reset(new Scene());
+    Scene::init();
     Loader::init(verbose, RESOURCE_DIR);
 
-    lastFrameTime = runTime = windowHandler.getTime();
+    lastFrameTime = runTime = Window::getTime();
    
     return 0;
 }
 
 void EngineApp::run() {
 
-    while (!windowHandler.shouldClose()) {
+    while (!Window::shouldClose()) {
         /* Update time and FPS */
-        runTime = windowHandler.getTime();
-        timeStep = (runTime - lastFrameTime);
+        runTime = Window::getTime();
+        timeStep = glm::min((runTime - lastFrameTime), 0.25); // right now physics/collisions break when dt is too large, so this is a workaround
```

Comment:
Hmmm - I don't like these hard constraints and magic numbers

Reply:
this is an unglamorous fix. I'll add a TODO to improve it, but this shouldn't even be a thing in the first place